src/bash/0001-Add-plugin-support-to-bash.patch

```diff
@@ -0,0 +1,222 @@
+ 	opt_multibyte=yes opt_cond_regexp=no opt_coproc=no
+ 	opt_casemod_attrs=no opt_casemod_expansions=no opt_extglob_default=no
+ 	opt_globascii_default=yes
++	opt_bash_plugin=no
+ fi
+ 
+ # Check whether --enable-alias was given.
+@@ -3197,6 +3201,11 @@ if test "${enable_xpg_echo_default+set}" = set; then :
+   enableval=$enable_xpg_echo_default; opt_xpg_echo=$enableval
+ fi
+ 
++# Check whether --enable-bash-plugin was given.
++if test "${enable_bash_plugin+set}" = set; then :
++  enableval=$enable_bash_plugin; opt_bash_plugin=$enableval
++fi
++
+ 
+ # Check whether --enable-mem-scramble was given.
+ if test "${enable_mem_scramble+set}" = set; then :
+diff --git a/configure.ac b/configure.ac
+index 52b4cdb..2d73d90 100644
+--- a/configure.ac
++++ b/configure.ac
+@@ -185,6 +185,7 @@ opt_dircomplete_expand_default=no
+ opt_globascii_default=yes
+ opt_function_import=yes
+ opt_dev_fd_stat_broken=no
++opt_bash_plugin=yes
+ 
+ dnl options that affect how bash is compiled and linked
+ opt_static_link=no
+@@ -206,6 +207,7 @@ if test $opt_minimal_config = yes; then
+ 	opt_multibyte=yes opt_cond_regexp=no opt_coproc=no
+ 	opt_casemod_attrs=no opt_casemod_expansions=no opt_extglob_default=no
+ 	opt_globascii_default=yes
++	opt_bash_plugin=no
+ fi
+ 
+ AC_ARG_ENABLE(alias, AC_HELP_STRING([--enable-alias], [enable shell aliases]), opt_alias=$enableval)
+@@ -245,6 +247,7 @@ AC_ARG_ENABLE(single-help-strings, AC_HELP_STRING([--enable-single-help-strings]
+ AC_ARG_ENABLE(strict-posix-default, AC_HELP_STRING([--enable-strict-posix-default], [configure bash to be posix-conformant by default]), opt_strict_posix=$enableval)
+ AC_ARG_ENABLE(usg-echo-default, AC_HELP_STRING([--enable-usg-echo-default], [a synonym for --enable-xpg-echo-default]), opt_xpg_echo=$enableval)
+ AC_ARG_ENABLE(xpg-echo-default, AC_HELP_STRING([--enable-xpg-echo-default], [make the echo builtin expand escape sequences by default]), opt_xpg_echo=$enableval)
++AC_ARG_ENABLE(bash-plugin, AC_HELP_STRING([--enable-bash-plugin], [enable bash plugin features]), opt_bash_plugin=$enableval)
+ 
+ dnl options that alter how bash is compiled and linked
+ AC_ARG_ENABLE(mem-scramble, AC_HELP_STRING([--enable-mem-scramble], [scramble memory on calls to malloc and free]), opt_memscramble=$enableval)
+@@ -263,6 +266,10 @@ dnl opt_readline and opt_history are handled later, because AC_PROG_CC needs
+ dnl to be run before we can check the version of an already-installed readline
+ dnl library
+ 
++
++if test $opt_bash_plugin = yes; then
++AC_DEFINE(BASH_PLUGIN)
++fi
+ if test $opt_alias = yes; then
+ AC_DEFINE(ALIAS)
+ fi
+diff --git a/execute_cmd.c b/execute_cmd.c
+index 8b3c83a..c83efea 100644
+--- a/execute_cmd.c
++++ b/execute_cmd.c
+@@ -5458,6 +5458,14 @@ execute_disk_command (words, redirects, command_line, pipe_in, pipe_out,
+ 	 leave it there, in the same format that the user used to
+ 	 type it in. */
+       args = strvec_from_word_list (words, 0, 0, (int *)NULL);
++
++#if defined (BASH_PLUGIN)
```

Comment:
>BASH_PLUGIN [](http://example.com/codeflow?start=14&length=11)

The name is too general, how about BASH_SHELL_EXECVE_PLUGIN?

Reply:
Fixed